tests/integration/schema.test.js

```diff
@@ -1,119 +1,259 @@
+import { u8aToHex, u8aToU8a } from '@polkadot/util';
 import { randomAsHex } from '@polkadot/util-crypto';
 
 import { DockAPI } from '../../src/api';
 
 import { createNewDockDID, createKeyDetail, getHexIdentifierFromDID } from '../../src/utils/did';
 import { FullNodeEndpoint, TestKeyringOpts, TestAccountURI } from '../test-constants';
 import { getPublicKeyFromKeyringPair } from '../../src/utils/misc';
-import { verifyCredential } from '../../src/utils/vc';
-import { DockBlobIdByteSize, createNewDockBlobId, blobHexIdToQualified } from '../../src/modules/blob';
+import { verifyCredential, verifyPresentation } from '../../src/utils/vc';
+import { blobHexIdToQualified, createNewDockBlobId, DockBlobIdByteSize } from '../../src/modules/blob';
 import Schema from '../../src/modules/schema';
 import VerifiableCredential from '../../src/verifiable-credential';
-import exampleCredential from '../example-credential';
 import exampleSchema from '../example-schema';
+import VerifiablePresentation from '../../src/verifiable-presentation';
+import getKeyDoc from '../../src/utils/vc/helpers';
+import DockResolver from '../../src/dock-resolver';
 
 let account;
 let pair;
 let publicKey;
 let dockDID;
 let keyDetail;
 let blobId;
-
-const vcInvalid = {
-  ...exampleCredential,
-  credentialSubject: {
-    id: 'invalid',
-    notEmailAddress: 'john.smith@example.com',
-    notAlumniOf: 'Example Invalid',
-  },
-};
+let keyDoc;
+let validCredential;
+let invalidCredential;
+let invalidFormatBlobId;
+let dockResolver;
 
 describe('Schema Blob Module Integration', () => {
-  const dock = new DockAPI();
-  let invalidFormatBlobId;
+  const dockApi = new DockAPI();
 
   // Generate first key with this seed. The key type is Sr25519
   const firstKeySeed = randomAsHex(32);
 
   beforeAll(async (done) => {
-    await dock.init({
+    await dockApi.init({
       keyring: TestKeyringOpts,
       address: FullNodeEndpoint,
     });
-    account = dock.keyring.addFromUri(TestAccountURI);
-    dock.setAccount(account);
-    pair = dock.keyring.addFromUri(firstKeySeed);
+    account = dockApi.keyring.addFromUri(TestAccountURI);
+    dockApi.setAccount(account);
+    pair = dockApi.keyring.addFromUri(firstKeySeed);
     publicKey = getPublicKeyFromKeyringPair(pair);
     dockDID = createNewDockDID();
     keyDetail = createKeyDetail(publicKey, dockDID);
-    await dock.sendTransaction(dock.did.new(dockDID, keyDetail));
+    await dockApi.sendTransaction(dockApi.did.new(dockDID, keyDetail));
     blobId = randomAsHex(DockBlobIdByteSize);
 
-    vcInvalid.credentialSchema = {
-      id: blobId,
-      type: 'JsonSchemaValidator2018',
-    };
-
     // Write invalid format blob
     invalidFormatBlobId = randomAsHex(DockBlobIdByteSize);
-    await dock.sendTransaction(dock.blob.new({
+    await dockApi.sendTransaction(dockApi.blob.new({
       id: invalidFormatBlobId,
-      blob: stringToHex('hello world'),
+      blob: u8aToHex(u8aToU8a('hello world')),
```

Comment:
looks like a bad merge here, should use `stringToHex`

Reply:
```suggestion
      blob: stringToHex('hello world'),
```